Sources/ApolloCodegenLib/Templates/SelectionSetTemplate.swift

```diff
@@ -6,17 +6,20 @@ struct SelectionSetTemplate {
   let isMutable: Bool
   let generateInitializers: Bool
   let config: ApolloCodegen.ConfigurationContext
+  let accessControlRenderer: () -> String
```

Comment:
I'm not a huge fan of this and would appreciate any other ideas on a solution.

tl;dr - `TemplateRenderer` holds the information necessary to determine what the correct access control string should be. `SelectionSetTemplate` does not conform to `TemplateRenderer` though so it needs a way to have it's 'parent' template provide the access control string when needed. You'll find the same pattern on `OperationDefinitionTemplate`.

Reply:
Consider renaming to `renderAccessControl()`